http4s/server/src/main/scala/endpoints/http4s/server/JsonEntities.scala

```diff
@@ -18,10 +18,19 @@ trait JsonEntitiesFromCodecs
     extends algebra.JsonEntitiesFromCodecs
     with EndpointsWithCustomErrors {
 
+  /* Setting `strict = true` means that this won't accept requests that are
+   * missing their Content-Type header. However, if we use `strict = false`,
+   * requests with incorrect specified `Content-Type` still get accepted.
+   */
   def jsonRequest[A](implicit codec: JsonCodec[A]): RequestEntity[A] =
-    req => {
+    req =>
       http4s.EntityDecoder
-        .decodeString(req)
+        .decodeBy(MediaType.application.json) { msg: http4s.Media[Effect] =>
```

Comment:
Dotty requires that we put parens here:

```suggestion
        .decodeBy(MediaType.application.json) { (msg: http4s.Media[Effect]) =>
```